### PR TITLE
Convert to js

### PR DESCRIPTION
**Pull request by stepansergeevitch**
Converted docker action to js because docker action only supports linux runners.
Basically runs the same scripts but from a js runtime

### CODE REVIEW

File: cleanup.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    'FIREBOLT_DEFAULT_REGION': core.getInput('region')	     
+}
+
+
+function resolve_local_file(file_path) {
+        return path.join(__dirname, file_path)
+}
+
+function stop_all(db_name, on_success, on_error) {
+    python_bin = path.join(core.getState('python_path'), 'python');
```
Comment (ptiurin) [inline cleanup.js:17]: Does python here alias python3?

**Review by ptiurin — APPROVED** [review]
Apart from the question above, LGTM!

File: .gitignore [hunk 2]
```diff
@@ -282,4 +280,4 @@ cython_debug/
 
 # Emacs
 *~
-\#*\#
\ No newline at end of file
+\#*\#
```
Comment (Jeseeq) [inline .gitignore:283]: you need to add `node_modules/` to gitignore
and run `npm install` in action.yml

File: cleanup.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+const core = require('@actions/core');
```
Comment (Jeseeq) [inline cleanup.js:1]: JavaScript convention usually is to use 2 spaces instead of tabs

File: setup.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+	return path.join(__dirname, file_path)
+}
+
+function setup_virtualenv(on_success, on_error) {
+	exec('python -m pip install virtualenv && python -m virtualenv ' + resolve_local_file('.venv'),
+		function(error, stdout, stderr) {
+			if (error != null) {
+				return on_error(error.message)
+			}
+			python_dir = path.join(resolve_local_file('.venv'), process.platform == 'win32' ? '/Scripts' : '/bin/');
```
Comment (Jeseeq) [inline setup.js:25]: `python_dir` is not defined here, I think.
it should be `const python_dir`

**Review by Jeseeq — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 2 above)
Comment (stepansergeevitch) [reply .gitignore:283]: I don't think that actions allow this. The only entrypoint that I can provide is a single js file.
Also, adding `node_modules` is recommended by github docs
https://docs.github.com/en/actions/creating-actions/creating-a-javascript-action#commit-tag-and-push-your-action-to-github

Comment (stepansergeevitch) [reply .gitignore:283]: They do
`git add action.yml index.js node_modules/* package.json package-lock.json README.md`

File: cleanup.js (hunk 1 above)
Comment (stepansergeevitch) [reply cleanup.js:17]: Yes, because it's a binary from a virtual environment

File: .gitignore (hunk 2 above)
Comment (Jeseeq) [reply .gitignore:283]: and then a few steps after they said
```Checking in your node_modules directory can cause problems. As an alternative, you can use a tool called @vercel/ncc to compile your code and modules into one file used for distribution.```
also we can use `rollup` to bundle this script and it dependencies to one file

Comment (Jeseeq) [reply .gitignore:283]: am not sure how it will be when we have to update some of dependencies, I've never commit `node_modules`

**Review by Jeseeq — APPROVED** [review]

**Comment by stepansergeevitch** [comment]
Integration tests pass for SDK
https://github.com/firebolt-db/firebolt-python-sdk/runs/4483136037?check_suite_focus=true

**Pull request merged**